src/spiceypy/spiceypy.py

```diff
@@ -285,6 +285,66 @@ def cell_time(cell_size) -> SpiceCell:
     return stypes.SPICETIME_CELL(cell_size)
 
 
+@contextmanager
+def KernelPool(kernels: list):
+    """
+    Context manager for SPICE kernels.
+
+    A function called within KernelPool will only have access to
+    the list of kernels with which the context manager was initialized.
+
+    Previously furnished kernels are temporarily unloaded while using the
+    context manager, but can still be accessed outside of it.
+
+    Example:
+
+    ```
+        spiceypy.furnsh("A", "B")
+
+        function_1()
+
+        with spiceypy.KernelPool(["A", "C", "D"]):
+
+            function_2()
+
+        function_3()
+    ```
+
+    In this example, functions 1, and 3 will have access to kernels A, and
+    B; while function 2 will have access to kernels A, C, and D.
+
+    :param kernels: List of (relative or absolute) paths to individual 
+    kernels and/or to meta-kernels files.
+
+    """
+
+    global_kernels = []
+
+    for i in range(ktotal("all")):
+
+        data = kdata(i, "all")
+
+        if data[1] == "META" or data[2] == "":
+
+            global_kernels.append(data[0])
+
+    local_kernels = kernels
```

Comment:
This used to be `self.local_kernels = kernels` in the version in which I defined the context manager as a class. Could have paid more attention while rewriting it. I would rename the parameter `kernels` as `local_kernels`, so that the code uses the same vocabulary as the docstring. Are you ok with that?

Reply:
yup local_kernels sounds good to me, but I think more work is needed to ensure a single str kernel file would work here. see https://github.com/AndrewAnnex/SpiceyPy/blob/56dec76843bee5097a43c5b07bb69c8890fb5514/src/spiceypy/spiceypy.py#L6170 where I check if the I get a single string or iterable, as strings are iterable themselves.

Reply:
Didn't know about the `Union` type. I added it and updated input-type tests.